### PR TITLE
Add bfs count islands

### PR DESCRIPTION
**Pull request by guswh11**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [X] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

- [X] **if done some changes :**
  - [X] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/bfs/count_islands.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Example 4:
+110011
+001100
+000001
+111100
+Answer: 5
+"""
+
+
+def counting_islands(grid):
```
Comment (keon) [inline algorithms/bfs/count_islands.py]: change the method name to `count_islands`

**Comment by keon** [comment]
looks good to me, but this code needs unittests

File: algorithms/bfs/count_islands.py (hunk 1 above)
Comment (guswh11) [reply algorithms/bfs/count_islands.py]: Thanks for your comment. I'll revise and reload it soon :)

File: algorithms/bfs/count_islands.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    for i in range(row):
+        for j, num in enumerate(grid[i]):
+            if num == 1 and visited[i][j] != 1:
+                visited[i][j] = 1
+                queue.append((i, j))
+                while len(queue) != 0:
+                    x = queue[0][0]
+                    y = queue[0][1]
+                    del queue[0]
```
Comment (keon) [inline algorithms/bfs/count_islands.py]: this can be shortened by
```
while queue:
    x, y = queue.pop(0)
```
though pop(0) is not very efficient.

**Comment by guswh11** [comment]
Things are changed. 
- Unittest is added. 
- Shortened the while-function.
- Method name is changed. (counting_islands -> count_islands)

**Pull request merged**